packages/apputils-extension/src/statusbarplugin.ts

```diff
@@ -188,25 +195,66 @@ export const runningSessionsStatus: JupyterFrontEndPlugin<void> = {
     if (settingRegistry) {
       let disposable: IDisposable;
       const onSettingsUpdated = (
-        settings: ISettingRegistry.ISettings
+        kernelSettings?: ISettingRegistry.ISettings,
+        terminalsSettings?: ISettingRegistry.ISettings
       ): void => {
-        const showStatusBarItem = settings.get('showStatusBarItem')
-          .composite as boolean;
+        const showTerminalsMap = {
+          'if-any': undefined,
+          never: false,
+          always: true
+        };
+        const showKernels =
+          (kernelSettings?.get('showStatusBarItem').composite as
+            | boolean
+            | undefined) ?? true;
+        const showTerminals =
+          showTerminalsMap[
+            (terminalsSettings?.get('showStatusBarItem').composite as
+              | 'if-any'
+              | 'never'
+              | 'always'
+              | undefined) ?? 'if-any'
+          ];
 
         disposable?.dispose();
-        if (showStatusBarItem) {
-          disposable = registerItem();
+        if (showKernels || showTerminals) {
+          disposable = registerItem({
+            showKernels,
+            showTerminals
+          });
         }
       };
 
-      void settingRegistry
-        .load('@jupyterlab/apputils-extension:sessions-settings')
-        .then(settings => {
-          onSettingsUpdated(settings);
-          settings.changed.connect(onSettingsUpdated);
-        });
+      const kernelsPluginId = '@jupyterlab/apputils-extension:kernels-settings';
+      const terminalPluginId = '@jupyterlab/terminal-extension:plugin';
+
+      void Promise.all([
+        // Settings may be missing if the respective plugins are not enabled/included.
+        kernelsPluginId in settingRegistry.plugins
```

Comment:
Can't we use `app.isPluginActivated(kernelsPluginId)` to know if the plugin is included and activated ?

Reply:
I think that `isPluginActivated` may be false for terminals initially because activation happens in order to resolve tokens, right? Since there is no token relation the order would be random.